src/resolver.rs

```diff
@@ -421,34 +421,56 @@ impl<'i> TypeAnnotator<'i> {
             AstStatement::Reference { name, .. } => {
                 let annotation = if let Some(qualifier) = ctx.qualifier.as_deref() {
                     // if we see a qualifier, we only consider [qualifier].[name] as candidates
-                    self.index
-                        .find_member(qualifier, name)
-                        .map(|v| to_variable_annotation(v, self.index))
+                    self.index.find_member(qualifier, name).map_or_else(
+                        || {
+                            try_to_implementation_annotation(
```

Comment:
i'm not a fan of this name.
I am open to discuss a namen-scheme that would indicate certain behavior (e.g. try_... are methods that return an Option)

But for now we dont do this, so i would stick with a normal find_implementation_annotation(...)

anyway it looks like there is a word missing? try_to_what?